src/Microsoft.AspNet.WebSockets.Protocol/CommonWebSocket.cs

```diff
@@ -592,11 +592,11 @@ private void ValidateSegment(ArraySegment<byte> buffer)
             }
             if (buffer.Offset < 0 || buffer.Offset > buffer.Array.Length)
             {
-                throw new ArgumentOutOfRangeException(nameof(buffer.Offset), buffer.Offset, string.Empty);
+                throw new ArgumentOutOfRangeException(nameof(buffer), buffer.Offset, string.Empty);
```

Comment:
The problem here is not merely the parameter name - it's that the message to the developer won't make much sense. The message will now say something like "The 'buffer' parameter is out of range. Current value: -123". However, that is clearly not true. It's the `Offset` *property* of the `buffer` parameter that is out of range.

This exception should probably not be an `ArgumentOutOfRangeException` at all. It should just be an `ArgumentException` saying that `The `Offset` property cannot be negative and cannot be greater than the size of the `Array` property's length.` and pass in `buffer` as the name of the argument.